crates/bevy_transform/src/transform_propagate_system.rs

```diff
@@ -2,75 +2,97 @@ use crate::components::{Children, GlobalTransform, Parent, Transform};
 use bevy_ecs::{
     entity::Entity,
     query::{Changed, With, Without},
-    system::Query,
+    system::{Local, Query},
 };
 
+/// Used for [`transform_propagate_system`]. Ignore otherwise.
+pub(crate) struct Pending {
+    parent: *const GlobalTransform,
```

Comment:
I'm confused. This just used `*const GlobalTransform` without comment?

Since you've added this method, you must have benchmarked that using an owned, cloned GlobalTransform here was significantly slower. But why not comment that there?

Fwiw, either this is significantly faster, or our thresholds for how much faster it should be our different.
I await the comment comparing the performance to judge which side I land on.

Reply:
I haven't tested the performance of it just yet, but this was mostly to keep the stack size the same as the recursive version. GlobalTransform as a value is 40 bytes, a pointer/reference is 4/8. This would introduce more cache misses and a parent with a large number of children will save a huge number of copies of the same GlobalTransform to the queue instead of using a reference/pointer to an unmoved parent.

If the difference isn't noticeable for very deep/large hierarchies, I'll gladly use the cloned version instead.

Reply:
Is the reason it fails because of the need to store it in a `Local`, or is there another reason besides that?

I don't know if there's a good way to re-use the backing storage across lifetimes.

Reply:
Half of it is indeed the cached local. The other half is that there's no way to detail a "valid for this iteration of the for loop over roots" that can be borrowed from within the DFS while loop without unsafe, at least as far as I can tell.

So long as the invariants that components cannot be moved around during the middle of a system execution, and parents are always updated before their children, this should remain both sound and correct.